opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/SingleFieldQuery.java

```diff
@@ -36,7 +38,9 @@ protected T createQueryBuilder(List<NamedArgumentExpression> arguments) {
         .orElseThrow(() -> new SemanticCheckException("'query' parameter is missing"));
 
     return createBuilder(
-        field.getValue().valueOf().stringValue(),
+        (field.getValue() instanceof LiteralExpression)
+            ? field.getValue().valueOf().stringValue()
+        : field.getValue().toString(),
```

Comment:
```suggestion
        : field.getValue().toString(),
```
Why not `getValue().valueOf().stringValue()`? As written, this line assumes a specific string representation of the value object, but [it can be anything](https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#toString--:~:text=Cloneable-,toString,-public%C2%A0String).

Reply:
Could it be anything there? If yes, why not to crash while using incorrect data?

Reply:
It actually should be `field.getValue().toString()` as `field.getValue().valueOf().stringValue()` on a qualified name gives a null pointer exception.

Reply:
How about `((ReferenceExpression)field.getValue()).getAttr()` ?